### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by kepta**
- Include `array` polyfill.
- I am a bit torn on API design of two different ways a user would want their custom loaders to be injected into the Webpack configuration. I have used the `webpackLoaders` inside the [`oneOf`](https://github.com/mapbox/underreact/blob/b6e2e2bcb13a9ea32f817fde92cf18fc6350cfce/lib/webpack-config/loaders-list.js#L47) property and `inclusiveWebpackLoaders` [here](https://github.com/mapbox/underreact/blob/b6e2e2bcb13a9ea32f817fde92cf18fc6350cfce/lib/webpack-config/loaders-list.js#L51). I see use cases for both, but I think it might be difficult for the user to understand which one to use.

@davidtheclark for review, please.

### CODE REVIEW

File: __tests__/env-vars.test.js [hunk 1]
```diff
@@ -108,7 +108,7 @@ describe('Setting DEPLOY_ENV should read the correct .env files', () => {
         mainFileContent = fs.readFileSync(mainFilename, 'utf-8');
       }));
 
-  test('.env.abc overrides .env file vars', () => {
+  test('.env.abc override .env file vars', () => {
```
Comment (davidtheclark) [inline __tests__/env-vars.test.js]: Shouldn't this be `overrides`?

**Comment by kepta** [comment]
Yes, I imagine an extension which might want to have a bunch of loaders. For example the brfs loader: 
```
{
      test: /\.js$/,
      include: [/node_modules\/dsv/],
      loader: 'transform-loader?brfs'
 }
```

**Comment by davidtheclark** [comment]
@kepta I don't think that you can add multiple rules for the same extension and have them all work. If a file matches the `test`, it will use only the loader chain defined in that rule; other rules will not apply. So I think that if you want to apply multiple rules to JS extensions, you'll need to include a rule in `webpackLoaders` that covers all the needs of JS files. (Unless all this changed with Webpack 4.)

File: __tests__/__snapshots__/no-config.test.js.snap [hunk 2]
```diff
@@ -32,14 +32,14 @@ Object {
...
-              "size": 97446,
+              "size": 137509,
               "type": "file",
             },
             Object {
               "extension": ".map",
               "name": "polyfill.chunk.js.map",
               "path": "<TEMP_DIR>/_underreact-site/underreact-assets/js/polyfill.chunk.js.map",
-              "size": 68299,
+              "size": 99002,
```
Comment (kepta) [inline __tests__/__snapshots__/no-config.test.js.snap]: The chunk size has changed due to addition of `fetch`.

**Comment by kepta** [comment]
Update on this PR:
- We no longer `inclusiveWebpackLoaders`. I will wait for issues/cases where injecting loaders to`oneOf`  is not sufficient and needs something similar to the previous approach of `inclusiveWebpackLoaders`.
- Replaced `entry.js` with `index.js` as the entry point to the app. I have done this to minimize the changes needed and keep in sync with the established norm of using `index.js` as an entry point.

File: lib/config/__snapshots__/urc.test.js.snap [hunk 3]
```diff
@@ -187,7 +187,7 @@ Urc {
   "publicAssetsPath": "underreact-assets",
   "publicDirectory": "/fake-volume/public",
   "rootDirectory": "/fake-volume",
-  "siteBasePath": "",
+  "siteBasePath": "/",
```
Comment (kepta) [inline lib/config/__snapshots__/urc.test.js.snap]: Keeping an empty `siteBasePath` creates a problem with SPA using HTML5 push.

File: lib/config/urc.js [hunk 4]
```diff
@@ -37,10 +38,8 @@ module.exports = class Urc {
   normalize() {
     const urc = this;
     // Normalize the merged configuration
-    if (urc.siteBasePath && urc.siteBasePath !== '/') {
-      urc.siteBasePath = urc.siteBasePath.replace(/\/$/, '');
-      urc.siteBasePath = path.join('/', urc.siteBasePath);
-    }
+    urc.siteBasePath = urlJoin('/', urc.siteBasePath);
```
Comment (davidtheclark) [inline lib/config/urc.js]: Are you sure that we'll end up with normalized `/` at the *end* of the string?

File: lib/config/validate-config.js [hunk 5]
```diff
@@ -45,6 +45,12 @@ function validateAbsolutePaths(value) {
   }
 }
 
+function validateBasePath(value) {
+  if (typeof value !== 'string' || value === '') {
+    return 'non-empty string';
+  }
+}
```
Comment (davidtheclark) [inline lib/config/validate-config.js]: Doesn't fusspot's `required` take care of checking for empty strings?

File: lib/config/urc.js (hunk 4 above)
Comment (kepta) [reply lib/config/urc.js]: I guess we don't need `urc.siteBasePath` to end with `/`. If at some place (Webpack's publicPath) we can append it there.

File: lib/config/validate-config.js (hunk 5 above)
Comment (kepta) [reply lib/config/validate-config.js]: I think fusspot's `required` takes care of `null` and `undefined`, an empty string is still a string so I don't think `required` would reject it.

**Pull request merged**